bentoml/utils/docker_utils.py

```diff
@@ -0,0 +1,185 @@
+        r"""
+        ^(
+        [a-z0-9]+      # alphanumeric
+        (.|_{1,2}|-+)? # seperators
+        )*$
+        """,
+        re.VERBOSE,
+    )
+    valid_version_pattern = re.compile(
+        r"""
+        ^
+        [a-zA-Z0-9] # cant start with .-
+        [ -~]{,127} # ascii match rest, cap at 128
+        $
+        """,
+        re.VERBOSE,
+    )
+
+    if not valid_name_pattern.match(name):
+        raise click.BadParameter(
+            f"Provided Docker Image tag {tag} is invalid. "
+            "Name components may contain lowercase letters, digits "
+            "and separators. A separator is defined as a period, "
+            "one or two underscores, or one or more dashes.",
+            ctx=ctx,
+            param=param,
+        )
+    if version and not valid_version_pattern.match(version):
+        raise click.BadParameter(
+            f"Provided Docker Image tag {tag} is invalid. "
+            "A tag name must be valid ASCII and may contain "
+            "lowercase and uppercase letters, digits, underscores, "
+            "periods and dashes. A tag name may not start with a period "
+            "or a dash and may contain a maximum of 128 characters.",
+            ctx=ctx,
+            param=param,
+        )
+    return tag
+
+
+def containerize_bento_service(
+    bento,
+    push=False,
+    tag=None,
+    build_arg=None,
+    username=None,
+    password=None,
+    pip_installed_bundle_path=None,
+):
+    """Containerize specified BentoService.
+
+    BENTO is the target BentoService to be containerized, referenced by its name
+    and version in format of name:version. For example: "iris_classifier:v1.2.0"
+
+    `bentoml containerize` command also supports the use of the `latest` tag
+    which will automatically use the last built version of your Bento.
+
+    You can provide a tag for the image built by Bento using the
+    `--docker-image-tag` flag. Additionally, you can provide a `--push` flag,
+    which will push the built image to the Docker repository specified by the
+    image tag.
+
+    You can also prefixing the tag with a hostname for the repository you wish
+    to push to.
+    e.g. `bentoml containerize IrisClassifier:latest --push --tag username/iris`
+    would build a Docker image called `username/iris:latest` and push that to
+    Docker Hub.
+
+    By default, the `containerize` command will use the credentials provided by
+    Docker. You may provide your own through `--username` and `--password`.
+    """
+    saved_bundle_path = resolve_bundle_path(bento, pip_installed_bundle_path)
+    _echo(f"bendle path is {pip_installed_bundle_path}")
+    _echo(f"bento is {bento}")
+    _echo(f"Found Bento: {saved_bundle_path}")
+
+    bento_metadata = load_bento_service_metadata(saved_bundle_path)
+    name = to_valid_docker_image_name(bento_metadata.name)
+    version = to_valid_docker_image_version(bento_metadata.version)
+
+    if not tag:
+        _echo(
```

Comment:
All of the `_echo` are related to CLI commands.  If we want to refactor this to general use, we need to remove them